solr/core/src/java/org/apache/solr/servlet/HttpSolrCall.java

```diff
@@ -1145,40 +1123,32 @@ protected String getRemoteCoreUrl(String collectionName, String origCorename)
           "No active replicas found for collection: " + collectionName);
     }
 
-    String coreUrl =
-        getCoreUrl(collectionName, origCorename, clusterState, activeSlices, byCoreName, true);
+    String coreUrl = getCoreUrl(origCorename, clusterState, activeSlices, byCoreName, true);
 
     if (coreUrl == null) {
-      coreUrl =
-          getCoreUrl(collectionName, origCorename, clusterState, activeSlices, byCoreName, false);
+      coreUrl = getCoreUrl(origCorename, clusterState, activeSlices, byCoreName, false);
     }
 
     return coreUrl;
   }
 
   private String getCoreUrl(
-      String collectionName,
       String origCorename,
       ClusterState clusterState,
-      List<Slice> slices,
+      Collection<Slice> slices,
       boolean byCoreName,
       boolean activeReplicas) {
     String coreUrl;
     Set<String> liveNodes = clusterState.getLiveNodes();
 
-    List<Slice> shuffledSlices;
-    if (slices.size() < 2) {
-      shuffledSlices = slices;
-    } else {
-      shuffledSlices = new ArrayList<>(slices);
-      Collections.shuffle(shuffledSlices, Utils.RANDOM);
-    }
+    Iterator<Slice> shuffledSlices = new RandomIterator<>(Utils.RANDOM, slices);
```

Comment:
I was wondering why we need to shuffle if we look for a specific core by its name. But I see that the byCoreName parameter is always false (byCoreName is now always false).
So the code wants to find any active replica of any shard in a specific collection, and then return a url with replica.getBaseUrl() + "/" + origCoreName. Do I understand correctly? Is it the intended behavior?

Reply:
Thanks for looking closer.  I see there's some dead code to remove -- byCoreName and don't pass origCorename in either.  I'll do another update to clarify.